pkg/bid/scanner.go

```diff
@@ -0,0 +1,90 @@
+package bid
+
+import (
+	"fmt"
+	"strings"
+)
+
+type tokenType int
+
+const (
+	colon   tokenType = iota
```

Comment:
I've just been reading about `iota` and it seems like the idiomatic way of writing these consts is the simplified: 
```go
const (
	colon tokenType = iota
	slash
	literal
	eof
)
```
In my local testing it does the same thing with the types and values.